src/Engine/GUI/UIPanelManager.cpp

```
@@ -139,68 +158,104 @@ void UIPanelManager::renderMenuBar() {
+				ImGuiUtils::AlignedText(ImGuiUtils::TEXT_ALIGN_MIDDLE, "Preparing...");
+				ImGui::EndPopup();
+			}
+		}
+
+
+		// Reset UI state before starting a new load
+		m_showLoadingModal = true; showInitialPopup = false;
+		m_currentLoadProgress = 0.0f;
+		m_currentLoadMessage = "Starting scene load...";
+		m_loadErrorOccurred = false;
+		m_loadErrorMessage = "";
```

Cppcheck (2.20.0)

[style] The function 'crossed' is never used.

(unusedFunction)